src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1556,8 +1557,22 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
     retry_exception = Exception(None)
     for _ in range(0, max_retry):
         try:
-            return sdk_no_wait(no_wait, client.create_or_update,
+           result = sdk_no_wait(no_wait, client.create_or_update,
                                resource_group_name=resource_group_name, resource_name=name, parameters=mc)
+           # add cluster spn with Monitoring Metrics Publisher role assignment to the cluster resource
+           if monitoring:
+            from msrestazure.tools import resource_id
+            cluster_resource_id = resource_id(
+                    subscription=subscription_id,
+                    resource_group=resource_group_name,
+                    namespace='Microsoft.ContainerService', type='managedClusters',
+                    name=name
+                )  
+            if not _add_role_assignment(cmd.cli_ctx, 'Monitoring Metrics Publisher',
+                                            service_principal_profile.client_id, scope=cluster_resource_id):
+                    logger.warning('Could not create a role assignment for monitoring addon. '
+                                'Are you an Owner on this subscription?')
```

Comment:
In RBAC, to grant access, you create a role assignment. To create a role assignment, use the Role Assignments - Create REST API and specify the security principal, role definition, and scope. To call this API, you must have access to the Microsoft.Authorization/roleAssignments/write operation. Of the built-in roles, only Owner and User Access Administrator are granted access to this operation.

https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-rest#grant-access

Reply:
Owner is this lower RBAC role which has  roleAssignments/write permission.